Api/Controllers/AttributesController.cs

```diff
@@ -17,33 +18,41 @@ public AttributesController(IAttributeGroupServices attributeServices, IGenericR
         _attributeRepo = attributeRepo;
     }
     [HttpGet]
-    public async Task<IActionResult> Get()
+    public async Task<IActionResult> GetAttributeGroups()
     {
         var attribute = await _attributeServices.List();
         return Ok(attribute);
     }
 
     [HttpGet("{id}")]
-    public async Task<IActionResult> Get(string id)
+    public async Task<IActionResult> GetAttributeGroup(string id)
     {
-        var attribute = _attributeServices.GetById(Guid.Parse(id));
-        return Ok(attribute.Result);
+        var attribute = await _attributeServices.GetById(Guid.Parse(id));
+        return Ok(attribute);
     }
 
     [HttpPost]
-    public async Task<IActionResult> Post([FromBody] AttributeGroupEntity attributeGroup)
+    public async Task<IActionResult> CreateAttributeGroup([FromBody] CreateAttributeGroupRequest attributeGroup)
     {
-        await _attributeServices.Create(attributeGroup);
-        return NoContent();
+        AttributeGroupEntity attGrpEnt = new AttributeGroupEntity();
+        Mapper.Map(attributeGroup,attGrpEnt);
+        await _attributeServices.Create(attGrpEnt);
+        return CreatedAtAction(nameof(GetAttributeGroup), new { id = attGrpEnt.Id }, attGrpEnt);
     }
 
     [HttpPut("{id}")]
-    public void Put(int id, [FromBody] string value)
+    public async Task<IActionResult> UpdateAttributeGroup(string id, [FromBody] UpdateAttributeGroupRequest attributeGroup)
     {
+        AttributeGroupEntity attGrpEnt = new AttributeGroupEntity();
+        Mapper.Map(attributeGroup, attGrpEnt);
+        await _attributeServices.Update(Guid.Parse(id), attGrpEnt);
+        return NoContent();
```

Comment:
return the updated entity to help FE easier when integrate with the APIs

Reply:
`return AcceptedAtAction(nameof(GetAttributeGroup), new { id = Guid.Parse(id) }, attGrpEnt);`

Reply:
return Ok

Reply:
return Ok